spring-pulsar/src/test/java/org/springframework/pulsar/core/ConsumerTestUtils.java

```diff
@@ -35,11 +45,20 @@ private ConsumerTestUtils() {
 	 * @param container container to spy on
 	 * @return the spied container object
 	 */
-	public static Consumer<?> spyOnConsumer(DefaultPulsarMessageListenerContainer<String> container) {
-		Consumer<?> consumer = getPropertyValue(container, "listenerConsumer.consumer", Consumer.class);
-		consumer = spy(consumer);
-		new DirectFieldAccessor(getPropertyValue(container, "listenerConsumer")).setPropertyValue("consumer", consumer);
-		return consumer;
+	@SuppressWarnings("unchecked")
+	public static Consumer<String> startContainerAndSpyOnConsumer(
```

Comment:
For some reason, when upgrading to 2.10.2, the former spyOnConsumer was resulting in a deadlock in `Consumer::batchReceive`.
This is an alternative that works better.